tests/test_generate_data.py

```diff
@@ -65,3 +66,79 @@ def test_params_same_from_sim_data(self):
+                                                          customer_id_col='customer_id', datetime_col='date',
+                                                          observation_period_end=observation_period_end, 
+                                                          freq='D')
+        summary_original['recency'] = summary_original['recency'].apply(np.ceil) # summary_data_from_transaction_data makes regency floor
+        summary_original = summary_original.reset_index(drop=True)
+        summary_test = summary_test.reset_index(drop=True)
+        assert summary_original.equals(summary_test)
+
+
+    # test using T as list
+    def test_beta_geometric_nbd_model_transactional_data_list_T(self):
+        bg_params_list = OrderedDict([('T', [400, 200, 5, 103, 198, 401]), ('r', 0.24), ('alpha', 4.41),  ('a', 0.79),
+                                 ('b', 2.43), ('size', 6)])
+        np.random.seed(188898)
+        summary_original = beta_geometric_nbd_model(**bg_params_list)[['frequency', 'recency', 'T']]
+        np.random.seed(188898)
+        observation_period_end = '2019-1-1'
+        summary_test = summary_data_from_transaction_data(transactions=beta_geometric_nbd_model_transactional_data(**bg_params_list),
+                                                          customer_id_col='customer_id', datetime_col='date',
+                                                          observation_period_end=observation_period_end, 
+                                                          freq='D')
+        summary_original['recency'] = summary_original['recency'].apply(np.ceil) # summary_data_from_transaction_data makes regency floor
+        summary_original = summary_original.reset_index(drop=True)
+        summary_test = summary_test.reset_index(drop=True)
+        assert summary_original.equals(summary_test)
+
+
+    # test using freq as hour
+    def test_beta_geometric_nbd_model_transactional_data_freq_hour(self):
+        np.random.seed(188898)
+        summary_original = beta_geometric_nbd_model(**bg_params)[['frequency', 'recency', 'T']]
+        np.random.seed(188898)
+        observation_period_end = '2019-1-1'
+        summary_test = summary_data_from_transaction_data(transactions=beta_geometric_nbd_model_transactional_data(**bg_params_with_freq),
+                                                          customer_id_col='customer_id', datetime_col='date',
+                                                          observation_period_end=observation_period_end, 
+                                                          freq='h')
+        summary_original['recency'] = summary_original['recency'].apply(np.ceil) # summary_data_from_transaction_data makes regency floor
+        summary_original = summary_original.reset_index(drop=True)
+        summary_test = summary_test.reset_index(drop=True)
+        assert summary_original.equals(summary_test)
+
+
+    # test using freq as week
```

Comment:
the function name gives a good idea of what the test is doing, you can remove these comments